src/fsharp/service/ServiceAssemblyContent.fs

```diff
@@ -186,14 +187,31 @@ type IAssemblyContentCache =
 module AssemblyContentProvider =
     open System.IO
 
-    let private createEntity ns (parent: Parent) (entity: FSharpEntity) =
+    let unresolvedSymbol (topRequireQualifiedAccessParent: Idents option) (cleanedIdents: Idents) (fullName: string) =
+        let getNamespace (idents: Idents) = 
+            if idents.Length > 1 then Some idents.[..idents.Length - 2] else None
+
+        let ns = 
+            topRequireQualifiedAccessParent 
+            |> Option.bind getNamespace 
+            |> Option.orElseWith (fun () -> getNamespace cleanedIdents)
+            |> Option.defaultValue [||]
```

Comment:
Can we use defaultWith or something?

Reply:
What do you mean? `[||]` is a singleton, no allocations here.

Reply:
It's not a singleton
```
> System.Object.ReferenceEquals([| |], [| |]);;
val it : bool = false
```
If you want to add a singleton table to array.fs `Array.empty` that would make sense, but make it a separate PR.  We'd also have to consider if `[| |]` should use that table.

Reply:
Hmm, interesting. I'll think about the table. Do you mean something like `Dictionary<Type, Type[]>`? Would lookup be 100% faster than allocating an empty array?

Reply:
answered in main comment section